server/controllers/userAuth.js

```diff
@@ -254,15 +255,15 @@ const recoveryPassword = (req, resp) => {
 
 const updatePassword = (req, resp) => {
   const { password: updatedPassword, passwordConfirmation } = req.body;
-  const { matches } = validator;
   const { token } = req.params;
 
   const sanitizedToken = sanitize(token);
-  const validationError = !matches(updatedPassword, /^[^\s]{4,32}$/);
-  const passwordsNoMatch =
-    _trim(updatedPassword) !== _trim(passwordConfirmation);
 
-  if (validationError || passwordsNoMatch) {
+  if (!validatePasswordStructure(updatedPassword)) {
+    return resp.sendStatus(400);
+  }
+
+  if (_trim(updatedPassword) !== _trim(passwordConfirmation)) {
```

Comment:
Such `ifs` can be combined again :) It won't evaluate second || argument if first is true.
So:
```
if (!validatePasswordStructure(updatedPassword) || _trim(updatedPassword) !== _trim(passwordConfirmation))
```